### ISSUE TITLE
CentOS fails to install

### ISSUE THREAD

**Issue by jjkola** [Magisk-Modules-Alt-Repo/chroot-distro#20]
> Please check report #9 for instructions on how to create a package zip. Please note, that you will have to reboot the device before the new version can be used.
> 
> I'll try to add instructions to documentation later but for now this will have to suffice. Also, I think it is time to create a new release shortly as there hasn't been any findings with the new code and it fixes several problems with original code.
> 
> If the issue still persist with latest code then please provide information about the exact rootfs you have downloaded so that I can try replicating the issue.

I installed the new module. At this point I was able to install and login in kali without errors.  When I try to log into CentOS I get the error /bin/su. Personally, I don’t need it, but I think it’s worth informing.

_Originally posted by @Dlazder in https://github.com/Magisk-Modules-Alt-Repo/chroot-distro/issues/6#issuecomment-2232868411_

**Comment by jjkola** [comment]
Investigated this issue and problem seems to be that there is no su/sudo in CentOS rootfs. This will need to be taken in account.

**Comment by YasserNull** [comment]
This is a problem, I can't testing all distributions

**Comment by jjkola** [comment]
It can be solved by adding automated testing. If done properly it can be used both for PR process as well as for daily/weekly/whatever automated run which checks that every supported distro can be downloaded and installed. For PR process we could add tests for known problems as well as general checks that things works as we expect them to work. This can be achieved for example with containers which has been tuned to look like a android device. And, I would assume that we are not the first ones needing this kind of thing, so I would expect to find examples of how to do this.

**Comment by jjkola** [comment]
While investigating the problem with adb shell I found out that when using termux the environment variables may be leaked to chrooted rootfs. This may have masked errors where rootfs is missing su, or is using su from wrong place (termux su). I have fixed this issue and I will try to push the changes as soon as possible. For the problem with su missing from the rootfs I have already thought a solution but will have to implement it.
